src/index.js

```diff
@@ -109,8 +109,17 @@ function optimist(fn) {
           return revertReducer(state, action);
       }
     }
-    let separated = separateState(state);
-    return baseReducer(separated.optimist, separated.innerState, action);
+    if (!state || !state.optimist || state.optimist.length) {
+      let separated = separateState(state);
+      return baseReducer(separated.optimist, separated.innerState, action);      
+    }
+    let nextState = fn(state, action);
```

Comment:
We should still be separating out the state, I think.  otherwise the `fn` gets called with a different value depending on whether there is a transaction in progress. My suggestion would be to do:

```js
const separated = separateState(state);
const nextState = fn(separated.innerState, action);
if (nextState === separated.innerState) {
  return state;
}
validateState(nextState, action);
return nextState;
```